scripts/util_functions.sh

```diff
@@ -318,10 +318,12 @@ loop_setup() {
 
 mount_apex() {
   $BOOTMODE || [ ! -d /system/apex ] && return
-  local APEX DEST
+  local APEX APEXJARS DEST FWK PATTERN SHCON
   setup_mntpoint /apex
   mount -t tmpfs tmpfs /apex -o mode=755
-  local PATTERN='s/.*"name":[^"]*"\([^"]*\).*/\1/p'
+  SHCON=$(cat /proc/self/attr/current)
+  echo "u:r:su:s0" > /proc/self/attr/current # work around LOS Recovery not allowing loop mounts in recovery context
```

Comment:
u:r:su:s0 may not exist

Reply:
It does. All recoveries use recovery for flashing zips and su for adb shell, which is why I had trouble diagnosing the issue: loop mounts always worked in adb shell but never in the flashing zip.

Reply:
https://review.lineageos.org/#/c/257053

Reply:
I think we make recovery permissive 4 years ago?

Reply:
All I know is shell with recovery context fails to do loop mounts, but change it to su context and it works. You can see the errors in the commit and log.

Reply:
userdebug_or_eng(`
permissive recovery;
')

recovery context is always permissive. Is there an avc log?

Reply:
I'll pull one. It's like this on any current Lineage device.

Reply:
Here is a fresh recovery.log of 26102 and the dmesg.log from the same session. Some parts say permissive=1 and some say permissive=0, which is interesting and may explain what I've been seeing and the workaround I found.

[recovery.log](https://github.com/topjohnwu/Magisk/files/11749729/recovery.log)
[dmesg.log](https://github.com/topjohnwu/Magisk/files/11749730/dmesg.log)

Reply:
I guess this might explain why Open GApps does `setenforce 0` as part of their installer! I'd never noticed it until now:

https://github.com/opengapps/opengapps/commit/a0832927c596153b2549a600061fd8ffeb67dc79

Reply:
Logs are clean of the mount errors in Permissive of course. So yeah, can either switch context like I suggested, or full SELinux Permissive for recovery installs. Whichever seems more appealing to @vvb2060, I suppose.

[recovery.log](https://github.com/topjohnwu/Magisk/files/11777244/recovery.log)
[dmesg.log](https://github.com/topjohnwu/Magisk/files/11777245/dmesg.log)

Reply:
Consider this resolved then since my workaround is viable? Or would you prefer full `setenforce 0` for recovery installs?